lib/activerecord-mysql-structure/active_record/structure_sql_sanitizer.rb

```diff
@@ -1,23 +1,89 @@
+# frozen_string_literal: true
+
 module ActiveRecordMySqlStructure
+  # This module is used by the enhanced rake task to sanitize the structure.sql
   class StructureSqlSanitizer
+    COLUMN_REGEX = /^\s*`\w+`.*,+$/
```

Comment:
Could use a stronger start and end check, and the `+` after `,` doesn't seem to be needed.

```suggestion
    COLUMN_REGEX = /\A\s*`\w+`.*,\z/

Reply:
the `+` is needed if there is a table that has no extra constraints (and no primary key).. Rare but possible.

I used `^` and `$` because I was looping on each line and don't really care about being strict across line breaks. However, I can make the change.

Reply:
I might just anchor it to the beginning of the line:

```
/\A\s*`\w+`/
```

Reply:
You mean it could have double commas like this?

```
`updated_at` datetime DEFAULT NULL,,
```

I tried it on my dev table:

```sql
DROP TABLE IF EXISTS `hendy`;
CREATE TABLE `hendy` (
  `id` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

Actually the comma might not be there, so yeah it should have been `,*` not `,+`, but that's useless. So your new regex is better.

Reply:
Yup, that's what I was thinking but got the regex wrong.